pkg/controller/seed-controller-manager/ipam/controller.go

```diff
@@ -89,10 +92,22 @@ func Add(
 	}
 
 	enqueueClustersForIPAMPool := handler.EnqueueRequestsFromMapFunc(func(a ctrlruntimeclient.Object) []reconcile.Request {
+		ctx := context.Background()
+
 		ipamPool := a.(*kubermaticv1.IPAMPool)
 
+		wasIPAMPoolDeleted, err := reconciler.checkIPAMPoolCleanup(ctx, ipamPool)
```

Comment:
I dont think that the IPAMPool cleanup should be happening in the EnqueueRequestsFromMapFunc, even though I understand why its convenient when the reconcile is working with Clusterrs, not IPAMPool.

The cleanup would be simpler if the IPAMPool controller would reconcile the IPAMPool, not sure why its doing Clusters and then listing IPAMPools?

Reply:
It's reconciling clusters because an IPAMPool can be applied before a cluster is created. So, in that case, when the cluster is created, the controller will allocate the subnet from the IPAMPool for it automatically.